### PR TITLE
Fix: Hide prechat for sessions inititated with setUser

### LINKED ISSUE #1894: Pre-chat form asks for name and email even when these are set in SDK
Hi there,
It appears that the pre-chat form still asks the user for their name and email address, even though these may have already been set via the SDK.

Name and Email already set in SDK:
Screen Shot 2021-03-10 at 8 43 26 AM

Pre-chat form should not ask for name and email if set in SDK:
Screen Shot 2021-03-10 at 8 53 07 AM

### PR DESCRIPTION
**Pull request by nithindavid**
Fixes #1894

### CODE REVIEW

File: app/javascript/widget/store/modules/contacts.js [hunk 1]
```diff
@@ -1,15 +1,36 @@
 import ContactsAPI from '../../api/contacts';
 import { refreshActionCableConnector } from '../../helpers/actionCable';
 
+const state = {
+  currentUser: {},
```
Comment (pranavrajs) [inline app/javascript/widget/store/modules/contacts.js:5]: What happens if the user reloads the page, `setUser` would be called only once if there is no change in the data.

File: app/controllers/api/v1/widget/contacts_controller.rb [hunk 2]
```diff
@@ -1,4 +1,9 @@
 class Api::V1::Widget::ContactsController < Api::V1::Widget::BaseController
+  def index
+    process_hmac
+    render json: @contact
```
Comment (pranavrajs) [inline app/controllers/api/v1/widget/contacts_controller.rb:4]: please pass only email and name

File: app/controllers/api/v1/widget/contacts_controller.rb [hunk 3]
```diff
@@ -1,4 +1,9 @@
 class Api::V1::Widget::ContactsController < Api::V1::Widget::BaseController
+  def index
+    process_hmac
```
Comment (pranavrajs) [inline app/controllers/api/v1/widget/contacts_controller.rb:3]: please move this as 

```
process_hmac before_action: [:index, :update]
```

File: config/routes.rb [hunk 4]
```diff
@@ -150,7 +150,7 @@
             post :transcript
           end
         end
-        resource :contact, only: [:update]
+        resource :contact, only: [:update, :index]
```
Comment (pranavrajs) [inline config/routes.rb:153]: Please order maintain the order index, update

File: app/javascript/widget/store/modules/contacts.js [hunk 5]
```diff
@@ -1,27 +1,58 @@
...
+};
+
+const SET_CURRENT_USER = 'SET_CURRENT_USER';
+const SET_CONTACTS_UI_FLAG = 'SET_CONTACTS_UI_FLAG';
+
+export const getters = {
+  getCurrentUser(_state) {
+    return _state.currentUser;
+  },
+  getUIFlags(_state) {
```
Comment (pranavrajs) [inline app/javascript/widget/store/modules/contacts.js:18]: UI flags is unused, please remove it.

File: app/javascript/widget/store/modules/contacts.js (hunk 1 above)
Comment (nithindavid) [reply app/javascript/widget/store/modules/contacts.js:5]: Added a case to fetch from API

**Pull request merged**